assets/js/Containers/FrevaGPT/index.js

```diff
@@ -85,37 +90,43 @@ class FrevaGPT extends React.Component {
       pathname: '/chatbot/',
       search: "",
     });
+    this.setState({ showSuggestions: true });
     window.scrollTo(0, 0)
   }
 
-  handleUserInput(e) {
-    this.setState({ userInput: e.target.value });
+  handleUserInput(e, callback) {
+    this.setState({ userInput: e.target.value }, callback);
   }
 
   async handleKeyDown(e) {
     if (e.key === "Enter") {
-      this.handleSubmit();
+      this.handleSubmit(e.target.value);
     }
   }
 
-  async handleSubmit() {
-    this.props.dispatch(addElement({ variant: "User", content: this.state.userInput}));
+  async submitUserInput() {
+    const userInput = this.state.userInput;
+    await this.handleSubmit(userInput);
     this.setState({ userInput: "" });
+  }
+
+  async handleSubmit(input) {
```

Comment:
It was seperated at one point but for the suggestion fields the state update is ansynchronous so calling both functions wouldn't work since the second one wouldn't receive the new state (which is not updated)